packages/semver/src/builders/version/plugin-handler.ts

```diff
@@ -0,0 +1,52 @@
+import { defer, Observable } from 'rxjs';
+
+import { Plugin, PluginDef, PluginMap, PluginOptions } from './plugin';
+import { _adapt } from './plugin-adapter';
+
+export class PluginHandler {
+  private _plugins: PluginMap;
+
+  constructor({ plugins }: { plugins: PluginDef[] }) {
+    this._plugins = _loadPlugins(plugins);
+  }
+
+  publish(): Observable<unknown> {
+    return this._run('publish');
+  }
+
+  private _run(hook: keyof Plugin): Observable<unknown> {
+    return defer(() => {
+      const plugins = this._plugins;
+      return Promise.all(
```

Comment:
I think we should run plugins sequentially as some plugins might have a side effect on others.